src/kOS.Safe/Screen/ScreenBuffer.cs

```diff
@@ -20,12 +22,16 @@ public interface IScreenBuffer
         void AddSubBuffer(SubBuffer subBuffer);
         void RemoveSubBuffer(SubBuffer subBuffer);
         List<char[]> GetBuffer();
+        void AddResizeNotifier(ScreenBuffer.ResizeNotifier notifier);
+        void RemoveResizeNotifier(ScreenBuffer.ResizeNotifier notifier);
+        string DebugDump();
     }
+    
 
     public class ScreenBuffer : IScreenBuffer
     {
-        public const int MAX_ROWS = 36;
-        public const int MAX_COLUMNS = 50;
+        private const int DEFAULT_ROWS = 36;
+        private const int DEFAULT_COLUMNS = 50;
         
         private int topRow;
```

Comment:
In order to understand some of the algorithm of ScreenBuffer, it's important to remember the meaning of this little "topRow" variable.  The ScreenBuffer is NOT just a buffer of the current terminal view, but it's actually the entire scrollback buffer of everything that got printed to the terminal in the past.  (It would be possible to use it to implement an actual vertical scrollbar for the terminal if we wanted to).  The topRow variable defines what portion of the ScreenBuffer is currently in view by defining which index into buffer is the first row currently shown on the screen.  (the bottom row is therefore at index (topRow + RowCount - 1). )